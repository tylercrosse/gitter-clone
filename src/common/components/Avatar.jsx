import React from 'react';
import { connect } from 'react-redux';
import { findOrCreateDirectMessage } from '../actions/';

export class Avatar extends React.Component {

  handleClick() {
    // ignore if not logged in;
    if (!this.props.loggedInUser.loggedIn) return;

    // ignore if loggedInUser = target

    // build props and fire action
    const creatorId = this.props.loggedInUser.id;
    const targetIds = [this.props.user.id];
    this.props.findOrCreateDirectMessage({ creatorId, targetIds });
  }
  render() {
    const { user } = this.props;
    return (
      <div onClick={(e) => this.handleClick(e)}>
        <img
        className="avatar"
        src={'http://i.pravatar.cc/30?u=' + user.name} alt={user.name}
        />
      </div>
    );
  }
}

Avatar.propTypes = {
  user: React.PropTypes.object.isRequired
};

const mapStateToProps = (state) => ({
  loggedInUser: state.user
});

export default connect(
  mapStateToProps,
  { findOrCreateDirectMessage }
)(Avatar);
