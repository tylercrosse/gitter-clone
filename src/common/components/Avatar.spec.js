import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Avatar } from './Avatar';

const setup = () => {
  const props = {
    user: {},
    loggedInUser: {},
    findOrCreateDirectMessage: jest.fn()
  };
  const component = <Avatar {...props} />;
  const wrapper = shallow(component);

  return {
    props,
    component,
    wrapper
  };
};

describe('<Avatar />', () => {
  it('should render correctly', () => {
    const { wrapper } = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
