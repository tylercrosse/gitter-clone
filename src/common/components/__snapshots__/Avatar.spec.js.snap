// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should render correctly 1`] = `
<div
  onClick={[Function]}
>
  <img
    className="avatar"
    src="http://i.pravatar.cc/30?u=undefined"
  />
</div>
`;
